loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts

```diff
@@ -23,59 +30,54 @@ export class EXT_meshopt_compression implements IGLTFLoaderExtension {
     public enabled: boolean;
 
     /**
-     * Path to decoder module; defaults to https://preview.babylonjs.com/meshopt_decoder.module.js
+     * Path to decoder module; defaults to https://preview.babylonjs.com/meshopt_decoder.js
      */
-    public static DecoderPath: string = "https://preview.babylonjs.com/meshopt_decoder.module.js";
+    public static DecoderPath: string = "https://preview.babylonjs.com/meshopt_decoder.js";
 
     private _loader: GLTFLoader;
-    private _decoder: Promise<any>;
+    private _decoderPromise?: Promise<any>;
 
     /** @hidden */
     constructor(loader: GLTFLoader) {
         this.enabled = loader.isExtensionUsed(NAME);
         this._loader = loader;
 
         if (this.enabled) {
-            var url = Tools.GetAbsoluteUrl(EXT_meshopt_compression.DecoderPath);
-
-            this._decoder = import(/* webpackIgnore: true */ url).then(function (result) {
+            this._decoderPromise = Tools.LoadScriptAsync(EXT_meshopt_compression.DecoderPath).then(() => {
                 // Wait for WebAssembly compilation before resolving promise
-                var MeshoptDecoder = result.MeshoptDecoder;
-                return MeshoptDecoder.ready.then(() => MeshoptDecoder);
+                return MeshoptDecoder.ready;
             });
         }
     }
 
     /** @hidden */
     public dispose() {
+        (this._loader as any) = null;
+        delete this._decoderPromise;
     }
 
     /** @hidden */
     public loadBufferViewAsync(context: string, bufferView: IBufferView): Nullable<Promise<ArrayBufferView>> {
```

Comment:
I noticed you removed the `return null;` section (which is great!!) this means that the return signature can be changed, and that this function can be converted to an async/await function (if it makes sense).

Reply:
The helper function `LoadExtensionAsync` will still return null.

Reply:
Also, we don't want to use async/await because we target es5.

Reply:
Typescript takes care of the conversion to es5. Use async await as much as you can

Reply:
actually one of the polyfill for generators is not in for ie (I guess it is only ie) that using async await would require if I remember well the issue.

Reply:
it also generates a whole ton of code (bad for code size) and is super hard to debug in js without the map file

Reply:
that was why we were not relying on it in the code.

Reply:
I wish we can use async/await. It would simplify the code a lot.

Reply:
OHHHHH YEESSSSS ME TOOO !!!!